contracts/hyphen/LiquidityPool.sol

```diff
@@ -388,105 +507,122 @@ contract LiquidityPool is
+    function sendFundsToUserFromCCMP(SendFundsToUserFromCCMPArgs calldata args) external whenNotPaused {
+        // CCMP Verification
+        (address senderContract, uint256 sourceChainId) = _ccmpMsgOrigin();
+        require(senderContract == chainIdToLiquidityPoolAddress[sourceChainId], "24");
 
-        require(!status, "Already Processed");
-        processedHash[hashSendTransaction] = true;
+        // Get local token address
+        address tokenAddress = tokenManager.symbolToTokenAddress(args.tokenSymbol, block.chainid);
+        require(tokenAddress != address(0), "25");
 
-        // uint256 amountToTransfer, uint256 lpFee, uint256 transferFeeAmount, uint256 gasFee
-        uint256[4] memory transferDetails = getAmountToTransfer(initialGas, tokenAddress, amount, tokenGasPrice);
+        // Calculate token amount on this chain
+        uint256 tokenDecimals = _getTokenDecimals(tokenAddress);
+        require(tokenDecimals & args.sourceChainDecimals != 0, "48");
+        uint256 amount = _getDestinationChainTokenAmount(
+            args.sourceChainAmount,
+            args.sourceChainDecimals,
+            tokenDecimals
+        );
 
-        liquidityProviders.decreaseCurrentLiquidity(tokenAddress, transferDetails[0]);
+        _verifyExitParams(tokenAddress, amount, args.receiver);
 
-        if (tokenAddress == NATIVE) {
-            (bool success, ) = receiver.call{value: transferDetails[0]}("");
-            require(success, "Native Transfer Failed");
-        } else {
-            SafeERC20Upgradeable.safeTransfer(IERC20Upgradeable(tokenAddress), receiver, transferDetails[0]);
+        (uint256 lpFee, uint256 incentivePoolFee, uint256 transferFeeAmount) = _calculateAndUpdateFeeComponents(
+            tokenAddress,
+            amount
+        );
+
+        // Calculate final amount to transfer
+        uint256 amountToTransfer;
+        require(transferFeeAmount <= amount, "28");
+        unchecked {
+            amountToTransfer = amount - (transferFeeAmount);
+        }
+
+        // Enforce that atleast minAmount (encoded in bytes as hyphenArgs[0]) is transferred to receiver
+        // This behaviour is overridable if they initiate the transaction from the ReclaimerEOA
+        if (args.hyphenArgs.length > 0) {
+            (uint256 minAmount, address reclaimerEoa) = abi.decode(args.hyphenArgs[0], (uint256, address));
+            require(
+                tx.origin == reclaimerEoa ||
+                    amountToTransfer >=
+                    _getDestinationChainTokenAmount(minAmount, args.sourceChainDecimals, tokenDecimals),
+                "49"
+            );
         }
 
+        // Refund the gas fee if user is initiating the transaction
+        if (tx.origin == args.receiver) {
+            amountToTransfer += _getDestinationChainTokenAmount(
+                args.gasFeePaidInTokenAmount,
+                args.sourceChainDecimals,
+                tokenDecimals
+            );
+        }
```

Comment:
No need to do this, because on the source chain the gas fee was given to the relayer address and not to liquidity pool. We can't use pool liquidity to refund the gas fee.

This refund should be done at Relayer Node level off-chain. Relayer should provide an option to the user to refund the gasFee in case user himself do the exit transaction.